File: layers/relulayer.py
from abc             import ABCMeta, abstractmethod

from utils.consts  import  *
from layer         import Layer
from activlayer    import ActivLayer
import nets.network

##########################################################
class ReluLayer(ActivLayer):
    #-----------------------------------------------------------------
    def __init__(self, layerName, ntwk, prev_layer):
        assert(isinstance(ntwk, nets.network.Network))
        assert(isinstance(prev_layer, Layer))
        super(ReluLayer, self).__init__(layerName, ntwk, prev_layer)

    #-----------------------------------------------------------------
    def __str__(self):
        baseLayer = self.gBaseLayerStr()
        return ("Relu " + baseLayer + self.gStateSizesStr())


    #-----------------------------------------------------------------
    def gLayerType(self):
        return LAYER_TYPE_ReLU

    #-----------------------------------------------------------------
    def gTypeStr(self):
        return "Relu"

    #-----------------------------------------------------------------
    def qPassThrough(self):
        return False

File: compiler/be/layers/src/relulayer.cpp
#include <sstream>

#include "utils/inc/datatype.hpp"
#include "nets/inc/network.hpp"
#include "layers/inc/relulayer.hpp"



namespace kcc {
namespace layers {

//--------------------------------------------------------
ReluLayer::ReluLayer(const Params& params, Layer* prev_layer)
    : ActivLayer(params, prev_layer)
{}



//--------------------------------------------------------
std::string
ReluLayer::gString() const
{
    std::string baseLayerStr = gBaseLayerStr();
    std::string retVal;
    retVal = retVal + gTypeStr() + " " + baseLayerStr + gStateSizesStr();
    return retVal;
}


//--------------------------------------------------------
const char*
ReluLayer::gTypeStrStatic()
{
    return "Relu";
}

//--------------------------------------------------------
const char*
ReluLayer::gTypeStr() const
{
    return gTypeStrStatic();
}

//--------------------------------------------------------
bool
ReluLayer::qPassThrough() const
{
    return false;
}

//--------------------------------------------------------
bool
ReluLayer::qReluLayer() const
{
    return true;
}

} // namespace layers
} // namespace kcc


